vector/src/main/java/im/vector/app/features/home/room/detail/composer/MessageComposerViewModel.kt

```diff
@@ -183,135 +183,176 @@ class MessageComposerViewModel @AssistedInject constructor(
-                            }
+                        is ParsedCommand.SendRainbow                       -> {
+                            val message = slashCommandResult.message.toString()
+                            state.rootThreadEventId?.let {
+                                room.replyInThread(
+                                        rootThreadEventId = state.rootThreadEventId,
+                                        replyInThreadText = slashCommandResult.message,
+                                        formattedText = rainbowGenerator.generate(message))
+                            } ?: room.sendFormattedTextMessage(message, rainbowGenerator.generate(message))
                             _viewEvents.post(MessageComposerViewEvents.SlashCommandResultOk())
                             popDraft()
                         }
-                        is ParsedCommand.SendRainbowEmote         -> {
-                            slashCommandResult.message.toString().let {
-                                room.sendFormattedTextMessage(it, rainbowGenerator.generate(it), MessageType.MSGTYPE_EMOTE)
-                            }
+                        is ParsedCommand.SendRainbowEmote                  -> {
+                            val message = slashCommandResult.message.toString()
+                            state.rootThreadEventId?.let {
+                                room.replyInThread(
+                                        rootThreadEventId = state.rootThreadEventId,
+                                        replyInThreadText = slashCommandResult.message,
+                                        msgType = MessageType.MSGTYPE_EMOTE,
+                                        formattedText = rainbowGenerator.generate(message))
+                            } ?: room.sendFormattedTextMessage(message, rainbowGenerator.generate(message), MessageType.MSGTYPE_EMOTE)
+
                             _viewEvents.post(MessageComposerViewEvents.SlashCommandResultOk())
                             popDraft()
                         }
-                        is ParsedCommand.SendSpoiler              -> {
-                            room.sendFormattedTextMessage(
-                                    "[${stringProvider.getString(R.string.spoiler)}](${slashCommandResult.message})",
-                                    "<span data-mx-spoiler>${slashCommandResult.message}</span>"
+                        is ParsedCommand.SendSpoiler                       -> {
+                            val text = "[${stringProvider.getString(R.string.spoiler)}](${slashCommandResult.message})"
+                            val formattedText = "<span data-mx-spoiler>${slashCommandResult.message}</span>"
+                            state.rootThreadEventId?.let {
+                                room.replyInThread(
+                                        rootThreadEventId = state.rootThreadEventId,
```

Comment:
same. or maybe using if/else will be clearer for all those cases

Reply:
Actually, it's not exactly the same. Current code with `?:` will also send message in current timeline as fallback (if null). Changing with if/else will not have the same behaviour

Reply:
I see, `replyInThread()` can return `null`. From a SDK user POV this is a bit weird, I would rather have an error in this case and not fallback to send the event in the regular timeline.

Reply:
Correct, this is supposed to never happen so I agree, will refactor it a bit